src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx

```diff
@@ -17,58 +17,61 @@
 //
 
 import * as React from 'react';
+import Validator from 'admin/services/Validation';
+import * as Immutable from 'seamless-immutable';
 import {LabeledInput} from 'admin/components/website/additionalComponents/LabeledInput';
 import {SwitchBox} from 'admin/components/website/additionalComponents/SwitchBox';
+import {TagList} from 'admin/components/website/additionalComponents/TagList';
 
-export interface IWebsiteOverrideSettingsContentSection {
+export interface IWebsiteOverrideSettings {
 	list: string[];
 	listPropName: string;
-	onSubmit: (data: any) => void;
-	checkValidation: (value: string, touched: boolean) => string;
+	isControllable?: boolean;
 	controlPropName?: string;
 	controlPropValue?: boolean;
+	label: JSX.Element | string;
+	ID: string;
+	value: string;
+	touched: boolean;
+	onSubmit: (data: any) => void;
+	onChange: (value:string, touched:boolean) => void;
 }
 
-export class WebsiteOverrideSettingsContentSection extends
-	React.Component <IWebsiteOverrideSettingsContentSection, any> {
+export class WebsiteOverrideSettings extends React.Component <IWebsiteOverrideSettings, any> {
+	private readonly validator : Validator;
 
 	constructor (props) {
 		super(props);
-		this.state = {
-			value: '',
-			touched: false,
-		};
+
+		this.validator = new Validator();
 
 		this.onDelete = this.onDelete.bind(this);
 		this.onSubmit = this.onSubmit.bind(this);
 		this.onKeyPress = this.onKeyPress.bind(this);
 		this.onEdit = this.onEdit.bind(this);
 		this.onToggleCheck = this.onToggleCheck.bind(this);
 		this.getValidationError = this.getValidationError.bind(this);
-		this.buildList = this.buildList.bind(this);
 	}
 
 	getValidationError() {
-		const {value, touched} = this.state;
-		return this.props.checkValidation(value, touched);
+		return this.checkValidation(this.props.controlPropName);
 	}
 
 	onDelete (index: number) {
 		const {list, listPropName} = this.props;
 		if (index >= 0) {
-			const overrideList = list;
+			const overrideList = Immutable(list).asMutable();
```

Comment:
You should work with this as Immutable object - try [this](https://facebook.github.io/immutable-js/docs/#/List/splice), for example